tests/mq_llm_engine/test_error_handling.py

```diff
@@ -164,7 +164,7 @@ async def test_failed_abort(tmp_socket):
                     sampling_params=SamplingParams(max_tokens=10),
                     request_id=uuid.uuid4()):
                 pass
-        assert "KeyError" in repr(execinfo.value)
+        assert "MQEngineDeadError" in repr(execinfo.value)
```

Comment:
I think this change might break the logic for this test?

IIUC, what was being done previously was that a `KeyError` was being raised in the engine, which then caused an `MQEngineDeadError`. The test then checks that the original `KeyError` is still referenced in the raised `MQEngineDeadError`.

I think it's still important for the original error to be surfaced once, is this change here intentional as part of not repeating the stack trace a bunch of times?

Reply:
Oh Wait... That is not what I mean neither. My intention was to put a `MQEngineBatchError`.  Did a quick local fix, and it worked as I excpected as well.

> I think it's still important for the original error to be surfaced once, is this change here intentional as part of not repeating the stack trace a bunch of times?

Totally agree. But thinking the system as whole, this error is logged at least once in this snippet:

```python
#repo/vllm/engine/multiprocessing/engine.py
    def start(self):
        try:
            try:
                logger.debug("Starting Startup Loop.")
                self.run_startup_loop()
                logger.debug("Starting heartbeat thread")
                self.heartbeat_thread.start()
                logger.debug("Starting Engine Loop.")
                self.run_engine_loop()
            except Exception as e:
                logger.exception(repr(e))
```

> is this change here intentional as part of not repeating the stack trace a bunch of times? 

The challenge here from the client side, previously when there is an error (in the batch) all requests receives the same exception and after propagate this exception to the server layer where they are logged a lot. Therefore, you request might receive an exception informing `KeyError` which is nothing related to it, but an exception raised by another test. That's the why, for now, I think might make sense change this test here.

Reply:
Ah, yeah I gotcha. We don't want to log the full stack trace here for the original `RAISED_ERROR` since it was already logged when that error actually happened. However, looking at the example server logs you posted, it does look like we are at least keeping the string repr of the original exception on the propagated exception here:
```
ERROR 10-02 13:00:45 launcher.py:111] MQEngineBatchError("A batch generation failed. Inspect the stacktrace to find the original error: Exception('Error in model execution (input dumped to /tmp/err_execute_model_input_20241002-130045.pkl): FORCED EXCEPTION')")
```
(the `FORCED EXCEPTION` is there from your original exception)

I probably would not change the `RAISED_ERROR` here to be a `MQEngineBatchError`, because this is the error that we are making the underlying `LLMEngine` raise, and it will never do that. I think instead here we should keep a different error (`KeyError` is probably still fine) and check that the message from it makes it in here.

This should work, right?

```
RAISED_ERROR = KeyError("foo")

...

assert "foo" in repr(execinfo.value)
```

Reply:
Just updated.

I went back to the original code:
```python
RAISED_ERROR = KeyError
```

I did that before because there are some lines that we have:

```python
with pytest.raises(RAISED_ERROR):
```

After carefully reading the tests again, I actually had to change only few lines (in other places) to preserve the original tests. I think now makes more sense, thank you for help me realize that!

> This should work, right?
>
> RAISED_ERROR = KeyError("foo")
> 
> ...
> 
> assert "foo" in repr(execinfo.value)

Almost, I printed this test and I got:
```
Engine loop is not running. Inspect the stacktrace to find the original error: KeyError().
```

In this test this KeyError is raised from a request that does no exist:
```python
# Trigger an abort on the client side.
# This request ID does not exist, and will cause the engine to error
await client.abort(request_id="foo")
```

Not sure why I had to change before. I reverted and the tests run fine.